google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystemProvider.java

```diff
@@ -130,6 +130,23 @@ public static void setStorageOptions(StorageOptions newStorageOptions) {
   }
 
   /**
+   * Changes the default configuration for every filesystem object created
+   * from here on, including via SPI. If null then future filesystem objects
+   * will have the factory default configuration.
+   *
+   * <p>This is meant to be done only once, at the beginning
+   * of some main program, in order to force all libraries to use some settings we like.
+   *
+   * <p>Libraries should never call this. If you're a library then, instead, create your own
+   * filesystem object with the right configuration and pass it along.
+   *
+   * @param newDefault new default CloudStorageConfiguration
+   */
+  public static void setDefaultCloudStorageConfiguration(CloudStorageConfiguration newDefault) {
```

Comment:
That'd be a different mechanism, yes. I think for that you can abuse `CloudStorageFileSystemProvider.setStorageOptions`